IPython/core/completer.py

```diff
@@ -2059,3 +2069,15 @@ def _complete(self, *, cursor_line, cursor_pos, line_buffer=None, text=None,
         self.matches = _matches
 
         return text, _matches, origins, completions
+        
+    def fwd_unicode_match(self, text:str) -> Tuple[str, list]:
+        # initial code based on latex_matches() method
+        slashpos = text.rfind('\\')
+        # if text starts with slash
+        if slashpos > -1:
+            s = text[slashpos+1:]
+            return s, [x for x in names if x.startswith(s)]
+        
+        # if text does not start with slash
+        else:
+            return u'', []
```

Comment:
There is one more issue if we run the test here. 
We can see that if `[x for x in names if x.startswith(s)]` is empty, then we should not return `s` but an empty string. 

This might be a bug somewhere else in IPython, but it breaks the test suite. I can fix it like so : 
```diff
@ completer.py:2076 @ def fwd_unicode_match(self, text:str) -> Tuple[str, list]:
         # if text starts with slash
         if slashpos > -1:
             s = text[slashpos+1:]
-            return s, [x for x in names if x.startswith(s)]
+            candidates = [x for x in names if x.startswith(s)]
+            if candidates:
+                return s, [x for x in names if x.startswith(s)]
+            else:
+                return '', ()

         # if text does not start with slash
         else:
-            return u'', []
+            return u'', ()
```

Reply:
Cool!

Reply:
I find it confusing sometimes how python manages a few string/character related things.